### PR TITLE
Fixes types annotation

### PR DESCRIPTION
**Pull request by datumbox**
This is based on the work of @zhiqwang at #3057 

Just made a few minor changes to avoid the postponed evaluation of annotations.

### CODE REVIEW

File: torchvision/models/detection/retinanet.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import torch
 import torch.nn as nn
 from torch import Tensor
-from torch.jit.annotations import Dict, List, Tuple
+from torch.jit.annotations import Dict, List, Tuple, Optional
```
Comment (fmassa) [inline torchvision/models/detection/retinanet.py:8]: nit: is this needed?

**Review by fmassa — APPROVED** [review]
Thanks a lot!

File: torchvision/models/detection/retinanet.py (hunk 1 above)
Comment (zhiqwang) [reply torchvision/models/detection/retinanet.py:8]: It is used in here

https://github.com/pytorch/vision/blob/8e244797efb64b2761fd61855fcac0123c10b804/torchvision/models/detection/retinanet.py#L459

Comment (fmassa) [reply torchvision/models/detection/retinanet.py:8]: Ok. Let's keep this here then, as when we move the `# type:` annotations to be inline the import will be necessary

**Pull request merged**